spec/rspec/matchers/built_in/include_spec.rb

```diff
@@ -764,6 +764,39 @@ def matches?(_)
       end
     end
 
+    describe "expect(hash).to include(key_matcher => value)" do
+      it "passes when the matcher matches a pair", :if => (RUBY_VERSION.to_f > 1.8) do
```

Comment:
I have little to no idea why `:if` is needed. Just copy and pasted it.

Reply:
I mean... does it fail without this?

Reply:
It seems that tests were limited to Ruby > 1.8 since in 1.8.7:
```ruby
/foo/ === :food # => false
```
while in later versions it is `true`.
Specs using symbolic keys and regexp natural matchers all fail.

Reply:
Well what is the downside of trying `to_s` on the keys, even if just for 1.8.7?

Reply:
Won't `expect({a: 1}).to include('a' => 1)` start passing for other versions for no reason?

Reply:
Hmm, I guess the problem is actually in `a_string_matching`, I mean half of me is wondering if it is even valid for that to pass for symbols...

Reply:
`a_string_matching` is a semantic alias to `match` which is:
>    # Given a `Regexp` or `String`, passes if `actual.match(pattern)`

So what probably has to be fixed is the spec that uses `a_string_matching` to match against symbols.

Reply:
Actually ok, I'm happy with that, change the alias out for match, and then its ok because its clear it is Ruby's match semantics at work, not us. (I'm just worried about giving the wrong impression).

Reply:
Done. Also fixed some neighbour examples with the same mismatch.